src/field_theory/splitting_field.lean

```diff
@@ -202,4 +208,192 @@ end
+      k.comp (algebra_map _ _) = j :=
+nat.rec_on n (λ α _ _ _ β _ j _, by exactI ⟨j, j.comp_id⟩) $ λ n ih α _ f hf β _ j hj, by exactI
+have hndf : f.nat_degree ≠ 0, by { intro h, rw h at hf, cases hf },
+have hfn0 : f ≠ 0, by { intro h, rw h at hndf, exact hndf rfl },
+let ⟨r, hr⟩ := exists_root_of_splits _ (splits_of_splits_of_dvd j hfn0 hj
+  (factor_dvd_of_nat_degree_ne_zero hndf)) (mt is_unit_iff_degree_eq_zero.2 f.irreducible_factor.1) in
+have hmf0 : map (adjoin_root.of f.factor) f ≠ 0, from mt (map_eq_zero _).1 hfn0,
+have hsf : splits (adjoin_root.lift j r hr) f.remove_factor,
+by { rw ← X_sub_C_mul_remove_factor _ hndf at hmf0, refine (splits_of_splits_mul _ hmf0 _).2,
+  rwa [X_sub_C_mul_remove_factor _ hndf, ← splits_id_iff_splits, map_map, adjoin_root.lift_comp_of,
+      splits_id_iff_splits] },
+let ⟨k, hk⟩ := ih f.remove_factor (nat_degree_remove_factor' hf) (adjoin_root.lift j r hr) hsf in
+⟨k, by rw [algebra_map_succ, ← ring_hom.comp_assoc, hk, adjoin_root.lift_comp_of]⟩
+
+theorem adjoin_roots (n : ℕ) : ∀ {α : Type u} [field α], by exactI
+  ∀ (f : polynomial α) (hfn : f.nat_degree = n),
+    algebra.adjoin α (↑(f.map $ algebra_map α $ splitting_field_aux n f hfn).roots :
+      set (splitting_field_aux n f hfn)) = ⊤ :=
+nat.rec_on n (λ α _ f hf, by exactI algebra.eq_top_iff.2 (λ x, subalgebra.range_le _ ⟨x, rfl⟩)) $
+λ n ih α _ f hfn, by exactI
+have hndf : f.nat_degree ≠ 0, by { intro h, rw h at hfn, cases hfn },
+have hfn0 : f ≠ 0, by { intro h, rw h at hndf, exact hndf rfl },
+have hmf0 : map (algebra_map α (splitting_field_aux n.succ f hfn)) f ≠ 0 := mt (map_eq_zero _).1 hfn0,
+by { rw [algebra_map_succ, ← map_map, ← X_sub_C_mul_remove_factor _ hndf, map_mul] at hmf0 ⊢,
+rw [roots_mul hmf0, map_sub, map_X, map_C, roots_X_sub_C, finset.coe_union, finset.coe_singleton,
+    algebra.adjoin_union, ← set.image_singleton, algebra.adjoin_algebra_map α (adjoin_root f.factor)
+      (splitting_field_aux n f.remove_factor (nat_degree_remove_factor' hfn)),
+    adjoin_root.adjoin_root_eq_top, algebra.map_top,
+    is_algebra_tower.range_under_adjoin α (adjoin_root f.factor)
+      (splitting_field_aux n f.remove_factor (nat_degree_remove_factor' hfn)),
+    ih, is_algebra_tower.subalgebra_comap_top] }
+
+end splitting_field_aux
+
+/-- A splitting field of a polynomial. -/
+def splitting_field (f : polynomial α) :=
+splitting_field_aux _ f rfl
+
+namespace splitting_field
+
+variables (f : polynomial α)
+
+instance : field (splitting_field f) :=
+splitting_field_aux.field _ _
+
+instance inhabited : inhabited (splitting_field f) := ⟨37⟩
+
+instance : algebra α (splitting_field f) :=
+splitting_field_aux.algebra _ _
+
+protected theorem splits : splits (algebra_map α (splitting_field f)) f :=
+splitting_field_aux.splits _ _ _
+
+variables [algebra α β] (hb : splits (algebra_map α β) f)
+
+/-- Embeds the splitting field into any other field that splits the polynomial. -/
+def lift : splitting_field f →ₐ[α] β :=
```

Comment:
Can you prove this for `is_splitting_field`?

Reply:
Future PR.